dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java

```diff
@@ -1021,38 +972,37 @@ public Knowledge visitSyncBarrier(Relation sync_bar) {
 
         @Override
         public Knowledge visitSyncFence(Relation sync_fen) {
-            Set<Tuple> may = new HashSet<>();
+            EventGraph may = new EventGraph();
             List<Event> fenceEventsSC = program.getThreadEventsWithAllTags(VISIBLE, FENCE, PTX.SC);
             for (Event e1 : fenceEventsSC) {
                 for (Event e2 : fenceEventsSC) {
-                    if (!exec.areMutuallyExclusive(e1, e2)) {
-                        may.add(new Tuple(e1, e2));
+                    if (e1.hasTag(Tag.PTX.SC) && e2.hasTag(Tag.PTX.SC) && !exec.areMutuallyExclusive(e1, e2)) {
+                        may.add(e1, e2);
```

Comment:
Same as above: you undid changes of #516. Please make sure that no changes get undone.

Reply:
Oops, good that you noticed it. Some weird automatic merging effect that I didn't catch. Fixed.